addons/hr_attendance/models/hr_attendance.py

```diff
@@ -706,11 +710,12 @@ def _cron_auto_check_out(self):
             domain=[
                 ('employee_id', 'in', to_verify.mapped('employee_id').ids),
                 ('check_in', '>', (fields.Datetime.now() - relativedelta(days=1)).replace(hour=0, minute=0, second=0)),
-                ('check_out', '!=', False)], groupby=['check_in:day', 'employee_id'], aggregates=['worked_hours:sum'])
+                ('check_out', '!=', False)], groupby=['employee_id', 'id'])
 
         mapped_previous_duration = defaultdict(lambda: defaultdict(float))
         for rec in previous_duration:
-            mapped_previous_duration[rec[1]][rec[0].date()] += rec[2]
+            for previous_attendance in rec[1]:
+                mapped_previous_duration[rec[0]][check_in_tz(previous_attendance).date()] += previous_attendance.worked_hours
```

Comment:
As discussed, I think we could have something more readable and explicit like this (the loop over attendances isn't necessary as we group per id + we shouldn't have more than one open attendance per employee anyway):
```py
for emp_id, attendance in previous_duration:
    mapped_previous_duration[emp_id][check_in_tz(attendance).date()] += attendance.worked_hours
```